### ISSUE TITLE
Add support for Gemini models

### ISSUE THREAD

**Issue by mnvsk97** [truefoundry/cognita#402]

**Comment by AbhishekRP2002** [comment]
hi @mnvsk97 if this is not taken up yet by anyone, I would love to take this up and contribute.
- I think for the first point we have to update `model_gateway.py` and `model_config.sample.yaml`
- for the second one, can u please explain a little more ?

**Comment by AbhishekRP2002** [comment]
any updates on this ? @mnvsk97 

cc: @chiragjn

**Comment by AbhishekRP2002** [comment]
Hi @mnvsk97 @chiragjn please lemme know if this **enhancement** is still in the roadmap. 
Would be super grateful if given the chance to contribute

**Comment by chiragjn** [comment]
Hey Abhishek, sorry I didn't get chance to reply earlier.
I am wondering, now that Gemini has a openai compatible API (https://ai.google.dev/gemini-api/docs/openai), is this still needed? 

---

I know accessing Gemini without API Key via some service account will need separate work

**Comment by AbhishekRP2002** [comment]
> Hey [@AbhishekRP2002](https://github.com/AbhishekRP2002) , sorry I didn't get chance to reply earlier. I am wondering, now that Gemini has a openai compatible API (https://ai.google.dev/gemini-api/docs/openai), is this still needed?
> 
> I know accessing Gemini without API Key via some service account will need separate work

yeah u are right actually, i think with Gemini models accessible using the OpenAI lib we don't need to make any changes.
 I thought in the direction of using gemini models either via Google AI Studio ( google.generativeai package ) or Vertex AI.
Apologies.
yeah for service account we need the GOOGLE_APPLICATIONS_CREDENTIALS with authentication via google auth.

**Comment by mnvsk97** [comment]
Hello @AbhishekRP2002, would you still like to pick this up? Please let us know if you need any help in getting started. Thanks.
